src/main/java/org/cobbzilla/s3s3mirror/comparisonstrategies/strategies/SizeAndSha256ComparisonStrategy.java

```diff
@@ -0,0 +1,33 @@
+package org.cobbzilla.s3s3mirror.comparisonstrategies.strategies;
+
+import lombok.SneakyThrows;
+import org.cobbzilla.s3s3mirror.Sha256;
+import org.cobbzilla.s3s3mirror.comparisonstrategies.S3Sha256Retriever;
+import org.cobbzilla.s3s3mirror.store.FileSummary;
+
+public class SizeAndSha256ComparisonStrategy extends SizeOnlyComparisonStrategy {
+    private final S3Sha256Retriever s3ShaReceiver;
+    public SizeAndSha256ComparisonStrategy(S3Sha256Retriever s3ShaRetriever) {
+        this.s3ShaReceiver = s3ShaRetriever;
+    }
+
+    @Override
+    public boolean sourceDifferent(FileSummary source, FileSummary destination) {
+        return super.sourceDifferent(source, destination) || shasAreDifferent(source, destination);
+    }
+
+    @SneakyThrows
```

Comment:
Is `@SneakyThrows` really required here? If it's really needed, please add a comment.

Reply:
Upon thinking about this further I don't see it as required anymore. If issues occur getting the SHA value we just error on the side of copying the file. The SneakyThrows has now been removed.